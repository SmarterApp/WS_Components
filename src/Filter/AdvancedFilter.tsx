import * as React from "react";
import {
  AdvancedFilterCategoryModel,
  FilterOptionModel,
  OptionTypeModel,
  FilterType
} from "./FilterModels";
import { BtnGroupOption } from "../Button/BtnGroupOption";
import { ToolTip } from "../ToolTip/ToolTip";

export interface AdvancedFilterProps extends AdvancedFilterCategoryModel {
  onFilterOptionSelect: (data?: FilterOptionModel) => void;
}

export class AdvancedFilter extends React.Component<AdvancedFilterProps, {}> {
  constructor(props: AdvancedFilterProps) {
    super(props);
  }

  renderAllBtnContainer() {
    const {
      filterOptions,
      onFilterOptionSelect,
      displayAllButton,
      disabled
    } = this.props;
    let allBtnContainer: JSX.Element | undefined;
    const anySelected = filterOptions.some(fo => fo.isSelected);
    if (displayAllButton) {
      allBtnContainer = (
        <BtnGroupOption
          onClick={onFilterOptionSelect}
          disabled={disabled}
          selected={!anySelected}
          label="All"
        />
      );
    }

    return allBtnContainer;
  }

  renderTags() {
    const {
      filterOptions,
      onFilterOptionSelect,
      disabled,
      emptyOptionsText
    } = this.props;
    const tags: JSX.Element[] = [];

    if (filterOptions.length > 0) {
      if (
        filterOptions[0].filterType === FilterType.TechnologyType ||
        filterOptions[0].filterType === FilterType.Calculator
      ) {
        filterOptions.sort((a, b) =>
          b.label.localeCompare(a.label, undefined, {
            usage: "sort",
            sensitivity: "variant"
          })
        );
        if (filterOptions[0].filterType === FilterType.Calculator) {
          filterOptions.reverse();
        }
      } else {
        filterOptions.sort((a, b) =>
          a.key.localeCompare(b.key, undefined, {
            usage: "sort",
            numeric: true,
            sensitivity: "base"
          })
        );
      }

      filterOptions.forEach((t, i) => {
        tags.push(
          <BtnGroupOption
            onClick={() => onFilterOptionSelect(t)}
            disabled={disabled}
            selected={t.isSelected}
            label={t.label}
            key={t.key}
          />
        );
      });
    } else {
      tags.push(<div key={0}>{emptyOptionsText || "No options."}</div>);
    }

    return tags;
  }

  render() {
    let text: JSX.Element | undefined;
    const { disabled, label, helpText } = this.props;
    if (helpText) {
      text = <p>{helpText}</p>;
    }
    // replace "-" with spaces, replace "." with nothing.
    const id = label.replace(/\ /g, "-").replace(/\./g, "");
    if (disabled) {
      // tslint:disable-next-line:no-null-keyword
      return null;
    }

    return (
      <div
        id={`${id}-filter`.toLocaleLowerCase()}
        className={"filter-selection"}
      >
        <div className="filter-container-header">
          <label>
            <ToolTip helpText={text} displayIcon={text !== undefined}>
              <span className="tooltip-label" info-label="true">
                {label}
              </span>
            </ToolTip>
          </label>
        </div>
        <div
          className="nested-btn-group btn-group-sm toggle-group vertical"
          data-toggle="buttons"
        >
          {this.renderAllBtnContainer()}
          <div className="btn-group filter-btn-group">{this.renderTags()}</div>
        </div>
      </div>
    );
  }
}
