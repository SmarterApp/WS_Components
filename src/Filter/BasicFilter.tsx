import "../Assets/Styles/basic-filter.less";
import * as React from "react";
import {
  BasicFilterCategoryModel,
  FilterOptionModel,
  OptionTypeModel
} from "./AdvancedFilterModel";

export interface BasicFilterProps extends BasicFilterCategoryModel {
  selectedHandler: (data?: FilterOptionModel) => void;
}

export class BasicFilter extends React.Component<BasicFilterProps, {}> {
  constructor(props: BasicFilterProps) {
    super(props);
  }

  renderTags() {
    const tags: JSX.Element[] = [];
    const classname = "";
    if (this.props.filterOptions) {
      this.props.filterOptions.forEach((t, i) => {
        if (this.props.type == OptionTypeModel.DropDown) {
          tags.push(
            <option key={t.key} value={t.key}>
              {t.label}
            </option>
          );
        } else if (this.props.type == OptionTypeModel.radioBtn) {
          tags.push(
            <label key={t.key}>
              <input
                checked={t.isSelected}
                type="radio"
                key={t.key}
                value={t.key}
                onChange={() => this.props.selectedHandler(t)}
              />
              {t.label}
            </label>
          );
        }
      });
    }
    return tags;
  }

  findFilterOption(key: string) {
    return this.props.filterOptions.find(fil => fil.key == key);
  }

  renderFilterBody() {
    let tag: JSX.Element;

    if (this.props.type == OptionTypeModel.DropDown) {
      tag = (
        <select
          onChange={e =>
            this.props.selectedHandler(this.findFilterOption(e.target.value))
          }
        >
          {this.renderTags()}
        </select>
      );
    } else {
      tag = <form>{this.renderTags()}</form>;
    }

    return <div className="basic-filter-body">{tag}</div>;
  }

  render() {
    return (
      <div
        id={(this.props.label + "-basic-filter").toLocaleLowerCase()}
        className="basic-filter-selection"
      >
        <label>{this.props.label}</label>
        {this.renderFilterBody()}
      </div>
    );
  }
}
