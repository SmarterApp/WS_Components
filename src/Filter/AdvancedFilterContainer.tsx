import * as React from "react";
import "../Styles/advanced-filter.less";
import { AdvancedFilter } from "./AdvancedFilter";
import {
  AdvancedFilterCategoryModel,
  AdvancedFiltersModel,
  FilterOptionModel,
  OptionTypeModel
} from "./AdvancedFilterModel";

export interface AdvancedFilterContainerProps {
  filterOptions: AdvancedFilterCategoryModel[];
  onClick: (selected: AdvancedFilterCategoryModel[]) => void;
  isNested?: boolean;
  pageTitle?: string;
}

export interface AdvancedFilterContainerState {
  expanded: boolean;
}

export class AdvancedFilterContainer extends React.Component<
  AdvancedFilterContainerProps,
  AdvancedFilterContainerState
> {
  constructor(props: AdvancedFilterContainerProps) {
    super(props);

    this.state = {
      expanded: this.props.isNested ? true : false
    };
  }

  handleClick = () => {
    this.setState({ expanded: !this.state.expanded });
  };

  onSelect(category: AdvancedFilterCategoryModel, option?: FilterOptionModel) {
    const allPressed = option === undefined && category.displayAllButton;
    if (category.disabled) {
      return;
    }

    let newFilters = this.props.filterOptions.slice();
    const categoryIndex = newFilters.indexOf(category);
    let options = newFilters[categoryIndex].filterOptions.slice();

    if (allPressed || !category.isMultiSelect) {
      options.forEach(o => (o.isSelected = false));
    }

    if (option) {
      const optionIdx = options.indexOf(option);
      options[optionIdx].isSelected = !option.isSelected;
    }

    newFilters[categoryIndex].filterOptions = options;
    this.props.onClick(newFilters);
  }

  resetFilters() {
    const newFilters = this.props.filterOptions.slice();
    newFilters.forEach(cat =>
      cat.filterOptions.forEach(fo => (fo.isSelected = false))
    );
    this.props.onClick(newFilters);
  }

  hasActiveFilterIndicators() {
    let active = false;
    this.props.filterOptions.forEach(fil => {
      if (!fil.disabled) {
        fil.filterOptions.forEach(opt => {
          if (opt.isSelected) {
            active = true;
          }
        });
      }
    });
    return active;
  }

  renderFilterIndicators() {
    const tags: JSX.Element[] = [];

    this.props.filterOptions.forEach(fil => {
      if (!fil.disabled) {
        fil.filterOptions.forEach(opt => {
          if (opt.isSelected) {
            tags.push(
              <div className="filter-indicator" key={fil.label + opt.key}>
                {opt.label}&nbsp;<span
                  onClick={() => this.onSelect(fil, opt)}
                  className="fa fa-times-circle fa-small"
                />
              </div>
            );
          }
        });
      }
    });

    return tags;
  }

  renderSelected() {
    return <div className="filter-status">{this.renderFilterIndicators()}</div>;
  }

  renderFilterBody() {
    const filterTags = this.props.filterOptions.map((fil, i) => {
      return (
        <AdvancedFilter
          key={i}
          {...fil}
          selectedHandler={opt => this.onSelect(fil, opt)}
        />
      );
    });

    return (
      <div
        className="filter-body"
        aria-live="polite"
        aria-relevant="additions removals"
      >
        {filterTags}
      </div>
    );
  }

  renderExpandButton() {
    const className = this.state.expanded
      ? "fa fa-chevron-down"
      : "fa fa-chevron-right";
    const buttonText = this.state.expanded ? "Collapse " : "Expand ";
    return (
      <div>
        {this.hasActiveFilterIndicators() ? (
          <button
            onClick={() => this.resetFilters()}
            className="filter-reset-btn"
          >
            Reset Filters
          </button>
        ) : null}
        <button
          onClick={() => this.handleClick()}
          className="filter-expand-btn"
        >
          {buttonText}
          <span className={className} />
        </button>
      </div>
    );
  }

  renderPageTitle() {
    if (this.props.pageTitle) {
      return (
        <h1>
          <span className="filter-page-title">{this.props.pageTitle}</span>
        </h1>
      );
    } else {
      return null;
    }
  }
  renderCollapsedFilterContainer = () => {
    const { filterOptions } = this.props;

    return (
      <div className="filter-sub-header-container">
      {this.renderPageTitle()}
        <div className="filter-advanced-filter-header">
          <div className="filter-advanced-filter-title">
            <h3>
              <span className="fa fa-tasks" />&nbsp;Advanced Filters
            </h3>
          </div>
          {this.renderExpandButton()}
        </div>
        {this.renderSelected()}
      </div>
    );
  };

  renderExpanded() {
    let content = null;
    if (this.state.expanded) {
      content = (
        <div className="advanced-filter-container-expanded">
          {this.renderFilterBody()}
        </div>
      );
    }
    return content;
  }

  render() {
    return (
      <div className="advanced-filter-container">
        {this.renderCollapsedFilterContainer()}
        {this.renderExpanded()}
      </div>
    );
  }
}
