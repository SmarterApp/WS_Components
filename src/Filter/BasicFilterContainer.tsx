import * as React from "react";
import "../Assets/Styles/basic-filter.less";
import { BasicFilterCategoryModel, FilterOptionModel } from "./FilterModels";
import { BasicFilter } from "./BasicFilter";

/**
 * Properties interface for the BasicFilterContainer component.
 * @interface BasicFilterContainerProps
 * @member {BasicFilterCategoryModel[]} filterCategories
 * @member {(selected: BasicFilterCategoryModel[]) => void} onUpdateFilter
 * @member {boolean} containsAdvancedFilter
 * @member {() => void} handleAdvancedFilterExpand
 * @member {string?} filterId
 */
export interface BasicFilterContainerProps {
  filterCategories: BasicFilterCategoryModel[];
  onUpdateFilter: (selected: BasicFilterCategoryModel[]) => void;
  containsAdvancedFilter: boolean;
  handleAdvancedFilterExpand: () => void;
  filterId?: string;
}
/**
 * State interface for the BasicFilterContainer component.
 * @interface BasicFilterContainerState
 * @member {boolean} expanded
 */
export interface BasicFilterContainerState {
  expanded?: boolean;
}
/**
 * The BasicFilterContainer is a menu of drop down lists and radio buttons
 * that filter Items based on BasicFilterCategoryModels and their corresponding
 * filter options. Can expand the Advanced filter as well if they are coupled together.
 * @class BasicFilterContainer
 * @extends {React.Component<BasicFilterContainerProps, BasicFilterContainerState>}
 */
export class BasicFilterContainer extends React.Component<
  BasicFilterContainerProps,
  BasicFilterContainerState
> {
  constructor(props: BasicFilterContainerProps) {
    super(props);
    this.state = {
      expanded: this.props.containsAdvancedFilter ? false : true
    };
    this.handleClick = this.handleClick.bind(this);
  }
  /**
   * Sets the isSelected property of the filter that the user selected
   * and sets the state of the parent component with the new filter categories.
   * @method onFilterSelect
   * @param {BasicFilterCategoryModel} category
   * @param {FilterOptionModel} [option]
   */
  onFilterSelect(
    category: BasicFilterCategoryModel,
    option?: FilterOptionModel
  ) {
    const index = this.props.filterCategories.indexOf(category);
    let { filterCategories } = this.props;
    if (!category.disabled) {
      if (option !== undefined) {
        let newOptions = filterCategories[index].filterOptions.slice();
        const optionIdx = filterCategories[index].filterOptions.indexOf(option);
        newOptions = filterCategories[index].filterOptions.map(opt => ({
          ...opt,
          isSelected: false
        }));

        newOptions[optionIdx].isSelected = !option.isSelected;
        filterCategories[index].filterOptions = newOptions;
        this.props.onUpdateFilter(filterCategories);
      }
    }
  }
  /**
   * Returns an array Basic Filter component for each of the filter categories
   * @method renderFilters
   */
  renderFilters() {
    const { filterCategories } = this.props;
    return filterCategories.map((fil, i) => {
      return (
        <BasicFilter
          key={i}
          {...fil}
          selectedHandler={opt => this.onFilterSelect(fil, opt)}
        />
      );
    });
  }

  /**
   * If the BasicFilter and AdvancedFilter Containers are used in conjunction with each other,
   * this method sets expands and collapses the AdvancedFilterContainer
   * @method handleClick
   */
  handleClick() {
    this.setState({ expanded: !this.state.expanded });
    this.props.handleAdvancedFilterExpand();
  }

  render() {
    const { filterCategories, containsAdvancedFilter, filterId } = this.props;
    const { expanded } = this.state;
    const id = filterId ? filterId : "";
    let advancedFilterButton = null;

    // if the component is being used in conjunction
    // with the AdvancedFilterContainer we handle expanding it here
    if (containsAdvancedFilter) {
      advancedFilterButton = (
        <div className="basic-filter-button-container">
          <div>Advanced Filters</div>
          <button
            className="btn btn-white filter-button"
            onClick={this.handleClick}
          >
            {expanded ? "Hide" : "Show"}&nbsp;
            <span className={`fa fa-chevron-${expanded ? "down" : "right"}`} />
          </button>
        </div>
      );
    }
    let className = "basic-filter-container";
    if (expanded) {
      className = "basic-filter-container-expanded";
    }

    return (
      <div id={id} className={className}>
        <div className="basic-filter">{this.renderFilters()}</div>
        {advancedFilterButton}
      </div>
    );
  }
}
