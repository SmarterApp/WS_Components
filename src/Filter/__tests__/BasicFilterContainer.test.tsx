import * as React from "react";
import { BasicFilterContainer } from "../BasicFilterContainer";
import { BasicFilter } from "../BasicFilter";
import { BasicFilterCategoryModel } from "../FilterModels";
import { mockBasicFilterCategories } from "mocks/Filter/mocks";
import { shallow } from "enzyme";

describe("BasicFilterContainer", () => {
  const AdvFilExpand = jest.fn();

  const updateFilter = (filterCategories: BasicFilterCategoryModel[]) => {
    wrapper.setProps({ filterCategories });
  };

  let wrapper = shallow(
    <BasicFilterContainer
      filterCategories={mockBasicFilterCategories}
      onUpdateFilter={updateFilter}
      containsAdvancedFilter={false}
      handleAdvancedFilterExpand={() => {}}
    />,
    { lifecycleExperimental: true }
  );

  let wrapper1 = shallow(
    <BasicFilterContainer
      filterCategories={mockBasicFilterCategories}
      onUpdateFilter={updateFilter}
      containsAdvancedFilter={true}
      handleAdvancedFilterExpand={AdvFilExpand}
    />,
    { lifecycleExperimental: true }
  );

  it("has proper number of BasicFilters", () => {
    expect(wrapper.findWhere(node => node.type() === BasicFilter)).toHaveLength(
      2
    );
  });

  it("can select dropdown filter options", () => {
    const options = ["default", 7, 56, 960];
    options.forEach(opt => {
      wrapper
        .findWhere(node => node.type() === BasicFilter)
        .at(0)
        .dive()
        .find("select")
        .simulate("change", { target: { value: opt } });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("can select radio button filter options", () => {
    const options = ["ELA", "Math"];
    let i = 0;
    options.forEach(opt => {
      wrapper
        .findWhere(node => node.type() === BasicFilter)
        .at(1)
        .dive()
        .find("form")
        .find("input")
        .at(i)
        .simulate("change", { target: { value: opt } });
      expect(wrapper).toMatchSnapshot();
      i++;
    });
  });

  it("resets filters on keyboard events", () => {
    const keyCodes = [0, 13, 32];
    keyCodes.forEach(key => {
      wrapper
        .findWhere(node => node.type() === BasicFilter)
        .at(0)
        .dive()
        .find("select")
        .simulate("keyDown", { keyCode: key });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("expands the advanced filter", () => {
    expect(wrapper1).toMatchSnapshot();
    wrapper1.setState({ expanded: false });
    wrapper1.findWhere(node => node.type() === "button").simulate("click");
    expect(wrapper1).toMatchSnapshot();
    wrapper1.findWhere(node => node.type() === "button").simulate("click");
    expect(wrapper1).toMatchSnapshot();
  });
});
