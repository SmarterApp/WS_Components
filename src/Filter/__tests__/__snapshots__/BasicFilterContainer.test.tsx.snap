// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicFilterContainer can select dropdown filter options 1`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select dropdown filter options 2`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select dropdown filter options 3`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select dropdown filter options 4`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select radio button filter options 1`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer can select radio button filter options 2`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer expands the advanced filter 1`] = `
<div
  className="basic-filter-container"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
  <div
    className="basic-filter-button-container"
  >
    <div>
      Advanced Filters
    </div>
    <button
      className="btn btn-white filter-button"
      onClick={[Function]}
    >
      Show
       
      <span
        className="fa fa-chevron-right"
      />
    </button>
  </div>
</div>
`;

exports[`BasicFilterContainer expands the advanced filter 2`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
  <div
    className="basic-filter-button-container"
  >
    <div>
      Advanced Filters
    </div>
    <button
      className="btn btn-white filter-button"
      onClick={[Function]}
    >
      Hide
       
      <span
        className="fa fa-chevron-down"
      />
    </button>
  </div>
</div>
`;

exports[`BasicFilterContainer expands the advanced filter 3`] = `
<div
  className="basic-filter-container"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
  <div
    className="basic-filter-button-container"
  >
    <div>
      Advanced Filters
    </div>
    <button
      className="btn btn-white filter-button"
      onClick={[Function]}
    >
      Show
       
      <span
        className="fa fa-chevron-right"
      />
    </button>
  </div>
</div>
`;

exports[`BasicFilterContainer resets filters on keyboard events 1`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer resets filters on keyboard events 2`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`BasicFilterContainer resets filters on keyboard events 3`] = `
<div
  className="basic-filter-container-expanded"
  id=""
>
  <div
    className="basic-filter"
  >
    <BasicFilter
      code="Grade"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "7",
            "label": "Grades 3-5",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "56",
            "label": "Grades 6-8",
          },
          Object {
            "filterType": "Grade",
            "isSelected": false,
            "key": "960",
            "label": "High School",
          },
        ]
      }
      key="0"
      label="Grade"
      selectedHandler={[Function]}
      type={3}
    />
    <BasicFilter
      code="Subject"
      disabled={false}
      filterOptions={
        Array [
          Object {
            "filterType": "Subject",
            "isSelected": false,
            "key": "ELA",
            "label": "English Literacy/ Arts",
          },
          Object {
            "filterType": "Subject",
            "isSelected": true,
            "key": "Math",
            "label": "Mathematics",
          },
        ]
      }
      key="1"
      label="Subjects"
      selectedHandler={[Function]}
      type={2}
    />
  </div>
</div>
`;
